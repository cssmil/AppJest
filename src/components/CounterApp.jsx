import PropTypes from 'prop-types'
import { useState } from 'react';

export const CounterApp = ({value, name}) => { 
  
  const [counter, setCounter] = useState(value)
  const handleAdd = () => setCounter( counter + 1 );
  const handleSub = () => setCounter( counter - 1 );
  const handleReseT = () => setCounter( value );
  
  return(
    <> 
      <h2>AppTest Jest {name}</h2>
      <h1 >{ counter }</h1>
      <button onClick={ handleAdd } >
        +1
      </button>
      <button onClick={ handleSub } >
        -1
      </button>
      <button aria-label='btn-reset'  onClick={ handleReseT } >
        reset
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired
}

CounterApp.defaultProps = {
  value: 1,
  name: "name test default"
}