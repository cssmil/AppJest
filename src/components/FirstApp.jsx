import PropTypes from 'prop-types'

export const FirstApp = ({title, subTitle, name}) => {
  return (
    <>
      <h1 className="mt-2"> { title } </h1>
      <hr />
      <p> { subTitle } </p>
      <p> { name } </p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  // name: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  title: "no tiene titulo",
  subTitle: "No tiene sub-titulo",
  name: "No tiene name"
}