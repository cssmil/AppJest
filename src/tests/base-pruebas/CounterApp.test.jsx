import { describe, test, expect } from '@jest/globals';

import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../../components/counterApp"

describe('Pruebas en el <CounterApp />', () => {

  const initialValue = 10;

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<CounterApp value={ initialValue } />)
    expect( container ).toMatchSnapshot();
  })
  
  test('debe mostrar el valor inicial de 100 <CounterApp value={ 100 } />', () => {
    render( <CounterApp value={initialValue} /> )
    expect(screen.getByText('10') ).toBeTruthy()
  })

  test('debe incrementar el boton +1 ', () => {
    render( <CounterApp value={initialValue} /> )
    fireEvent.click(screen.getByText('+1') )
    expect( screen.getByText('11')  ).toBeTruthy()
  })

  test('debe restar el boton -1 ', () => {
    render( <CounterApp value={initialValue} /> ) 
    fireEvent.click(screen.getByText('-1') )
    expect( screen.getByText('9')  ).toBeTruthy()
  })

  test('debe reiniciar el contador a su estado inicial ', () => {
    render( <CounterApp value={ initialValue } /> ) 
    
    fireEvent.click( screen.getByText('+1') )
    fireEvent.click( screen.getByText('+1') )
    fireEvent.click( screen.getByText('+1') )
    fireEvent.click( screen.getByText('reset') )

    //screen.debug(); 
    //expect(screen.getByText('10')).toBeTruthy()

    fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}))
    expect(screen.getByText(initialValue)).toBeTruthy()
  })

})