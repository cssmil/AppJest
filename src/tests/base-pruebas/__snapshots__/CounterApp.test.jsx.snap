// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h2>
    AppTest Jest 
    name test default
  </h2>
  <h1>
    10
  </h1>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
</div>
`;
